### PR TITLE
[5/x] Disable emitting assert for u32 cast

### PR DESCRIPTION
**Pull request by greenhat**
**This PR is stacked on the #227 and should be merged after it**

This PR temporarily disables emitting assertion on u32 cast to get past #225 that is caused by #174.

Do not merge this PR if #174 is already fixed.

### CODE REVIEW

File: codegen/masm/src/codegen/emit/int32.rs
Comment (bitwalker) [inline codegen/masm/src/codegen/emit/int32.rs:116]: I've looked into this issue, and the fix is to remove the call to `assert_unsigned_int32` from `cast` when casting from i32 to u32, from u32 to i32, and from i32 to u64; and replace it with a call to a new helper called `assert_i32`, see the implementation of `assert_i64` in `emit/int64.rs` for how to implement it.

The problem is that the `assert_unsigned_int32` helper is intended to assert that the sign bit of a 32-bit value is unset, _not_ to assert that the value is non-negative, which is how we are using it in `cast`.

All the other uses of `assert_unsigned_int32` are correct

**Review by bitwalker — CHANGES_REQUESTED** [review]
I've had a chance to look into this issue, and found the problem. I'd suggest we implement the correct fix in this PR, rather than closing this one and opening a new one, but I'll leave that up to you.

**Comment by bitwalker** [comment]
This can be closed in favor of #242 once it is merged

**Comment by greenhat** [comment]
Closing in favor of #242

File: codegen/masm/src/codegen/emit/int32.rs
Comment (greenhat) [reply codegen/masm/src/codegen/emit/int32.rs:116]: I have no found `assert_i32` in #242 so I made #250

**Pull request closed without merging**
